### ISSUE TITLE
Bug: scrm aborts with unexpected error

### ISSUE THREAD

**Issue by jeromekelleher** [scrm/scrm#70]
I get the following

$ scrm 50 10 -T -r 25263.0 63157501 -l 0 -seed 1 2 3 > /dev/null
Error: Can't find child node to replace
Try 'scrm --help' for more information.

This reproducible, and always happens for this random seed (although for other seeds it might not; I haven't put a lot of time info figuring out the frequency). I'm using scrm 1.5.0 on Debian wheezy, x86_64.

Let me know if I can help tracking this down.

**Comment by paulstaab** [comment]
Thanks for report this Jerome!

It seems this is happening after `doCompletePruning` is called every 100k recombinations, which also explains why we don't see this in small models:

```bash
scrm 10 1 -r 20000 2 -l 0 -seed 33
```

```
Segment Nr: 100001
Sequence position: 1.753284
* Recombination at height 23755.16 (above 0x7f442ed77f18)
* Cutting subtree below recombination 
* * New leaf of local tree: 0x7f442ed77eb0
* * New root of subtree: 0x7f442ed77b08
* * Done
╦ ╦ ╦ ╦ ╦ ╦  ╦ ╦   ╦ ╦  -       0 - [...]
║ ║ ║ ╚╦╝ ║  ║ ║   ║ ║  - 155.6472 - 0x7f442ed77900(0) 
║ ║ ║  ╚═╦╝  ║ ║   ║ ║  - 197.5562 - 0x7f442ed77b70(0) 
║ ║ ║    ║   ╚╦╝   ║ ║  - 477.2972 - 0x7f442ed77ca8(0) 
║ ║ ╚╦═══╝    ║    ║ ║  - 1390.796 - 0x7f442ed78050(0) 
║ ║  ║        ║  ┬ ║ ║  - 3445.372 - 0x7f442ed770e0(0) old 
║ ║  ║        ╚═╦┘ ║ ║  - 3893.184 - 0x7f442ed780b8(0) 
║ ║  ║          ║  ╚╦╝  - 4419.885 - 0x7f442ed77350(0) 
║ ╚╦═╝          ║   ║   - 5255.574 - 0x7f442ed77898(0) 
╚╦═╝            ║   ║   - 7565.719 - 0x7f442ed774f0(0) 
 ║              ╚═╦═╝   - 13237.22 - 0x7f442ed77f18(0) 
 ║          ┬     ╚═══╦ - 23755.16 - 0x7f442ed77eb0(0) 0x7f442ed77b08(0) 
 ╚═════════╦┘           -   32636 - 0x7f442ed77f80(0) 
* Starting coalescence
* * * PRUNING: Removing node 0x7f442ed77fe8 from tree (unneeded)
Error: Can't find child node to replace
```

**Comment by paulstaab** [comment]
Okay, figured it out. The complete pruning invalidates the cache I use for faster getting the contemporary lines at the local recombination event. Should be easy to fix.

**Comment by jeromekelleher** [comment]
Great, thanks @paulstaab.

**Comment by paulstaab** [comment]
Looking at it again, the algorithm is much better at removing old nodes from the tree than I though and there is no reason at all to do the complete pruning step. Just disabling that fixes the problem.

The fixed version should be in the master branch later today. I'll include the fix in version 1.6.0, which will be release with few weeks or so.

**Comment by gerton** [comment]
Relevant quote:  https://twitter.com/codinghorror/status/506010907021828096 <https://twitter.com/codinghorror/status/506010907021828096>
